### ISSUE TITLE
ChromHMM with additional genomes

### ISSUE THREAD

**Issue by kernco** [jernst98/ChromHMM#15]
ChromHMM's usability can be improved for users working with genomes that aren't included with the software. If ChromHMM is installed in a user's local directory, they can manually generate files and put them in the ANCHORFILES, COORDS, and CHROMSIZES directories. But for the purpose of running ChromHMM on a computing cluster or as part of a publicly distributed pipeline, users don't always have permission to write to the location that ChromHMM is installed and might not have the skills to set up their own local installation. It would help if the location of these files could be specified, and weren't required to be in the directory where ChromHMM is installed. Going a step further, it would be great if ChromHMM included commands to generate these files from a GTF/GFF file (COORDS and ANCHORFILES) and FASTA file (CHROMSIZES).

**Comment by jernst98** [comment]
Hi,

Thanks for the feedback. I don't think adding command line options to specify non-default ANCHROFILES, COORDS, and CHROMSIZES directories will be too difficult. I'll look into including that the next time there is a new ChromHMM release. 

Assuming Java is installed, then installing ChromHMM should just require unzipping a zip file, so hopefully most users should be able to do that if needed.

Are the genomes you are working with supported by the UCSC genome browser? If so you can use the fetchChromSizes utility to get the CHROMSIZES file. I have written a short Java program to convert gene table browser output from the UCSC gene browser into the format of COORDS and ANCHORFILES. I can look into the best way of distributing that.

Best,
-Jason

**Comment by jernst98** [comment]
In Version 1.16 I added options to the LearnModel command to specify non-default COORDS and ANCHORFILES directories, '-u coorddir' and '-v anchorfiledir' respectively. LearnModel previously supported specifying the chromosome length file with the '-l chromosomelengthfile' option.

I also added a main command to ChromHMM, ConvertGeneTable, to convert a gene table from the UCSC genome table browser into COORDS and ANCHORFILES gene annotation files.

-Jason
